### PR TITLE
Set initial temperature lower bound from EOS

### PR DESCRIPTION
**Pull request by ctrandrsn**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/PointwiseFunctions/AnalyticData/GrMhd/SpecInitialData.cpp [hunk 1]
```diff
@@ -100,47 +100,49 @@ template <size_t ThermodynamicDim>
...
   const size_t num_points = get_size(get(rest_mass_density));
   for (size_t i = 0; i < num_points; ++i) {
     const double local_rest_mass_density =
         get_element(get(rest_mass_density), i);
-    if (local_rest_mass_density <= density_cutoff) {
-      get_element(get(*specific_internal_energy), i) = 0.;
-    } else {
+    const double local_temperature =
+        get_element(get(temperature), i);
       if constexpr (ThermodynamicDim == 1) {
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/AnalyticData/GrMhd/SpecInitialData.cpp]: Can you format from line 113->127? I think these don't need to be indent as much

File: src/PointwiseFunctions/AnalyticData/GrMhd/SpecInitialData.cpp [hunk 2]
```diff
@@ -100,47 +100,49 @@ template <size_t ThermodynamicDim>
...
             get(eos.specific_internal_energy_from_density(
                 Scalar<double>(local_rest_mass_density)));
       } else if constexpr (ThermodynamicDim == 2) {
         get_element(get(*specific_internal_energy), i) =
             get(eos.specific_internal_energy_from_density_and_temperature(
-                Scalar<double>(local_rest_mass_density), Scalar<double>(0.)));
+                Scalar<double>(local_rest_mass_density),
+                Scalar<double>(local_temperature)));
       } else {
         ERROR("Only 1d & 2d EOS is currently supported for SpEC ID");
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/AnalyticData/GrMhd/SpecInitialData.cpp]: [FYI] These are the checks that we will want to replace to support a 3d EOS :)

File: src/PointwiseFunctions/AnalyticData/GrMhd/SpecInitialData.cpp [hunk 3]
```diff
@@ -100,47 +100,49 @@ template <size_t ThermodynamicDim>
...
   const size_t num_points = get_size(get(rest_mass_density));
   for (size_t i = 0; i < num_points; ++i) {
     const double local_rest_mass_density =
         get_element(get(rest_mass_density), i);
-    if (local_rest_mass_density <= density_cutoff) {
-      get_element(get(*pressure), i) = 0.;
-    } else {
+    const double local_temperature =
+        get_element(get(temperature), i);
       if constexpr (ThermodynamicDim == 1) {
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/AnalyticData/GrMhd/SpecInitialData.cpp]: Can you also run clang-format on lines 145-161?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
LGTM, a couple formatting suggestions/requests. Please squash immediately!

File: src/PointwiseFunctions/AnalyticData/GrMhd/SpecInitialData.cpp (hunk 3 above)
Comment (ctrandrsn) [reply src/PointwiseFunctions/AnalyticData/GrMhd/SpecInitialData.cpp]: I've run clang-format on the file so I think formatting should be fixed.

Comment (nilsdeppe) [reply src/PointwiseFunctions/AnalyticData/GrMhd/SpecInitialData.cpp]: Hmm, I would've expected line 103:
```cpp
    const double local_temperature =
        get_element(get(temperature), i);
      if constexpr (ThermodynamicDim == 1) {
```
to become
```cpp
    const double local_temperature =
        get_element(get(temperature), i);
    if constexpr (ThermodynamicDim == 1) {
```

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**